### PR TITLE
Offers tests

### PR DESCRIPTION
**Pull request by jonzee**
Restructured `Detail` component and added tests.

### CODE REVIEW

File: src/sass/Detail.scss [hunk 1]
```diff
@@ -77,7 +78,8 @@
     }
   }
 }
-.description {
+.detail-offer {
+  flex: 1;
```
Comment (michalmikolajczyk) [inline src/sass/Detail.scss]: what's this?

File: src/ts/Detail/DetailForm.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    const start = moment().toDate();
+    const wrapper = mountWithTheme(<DetailForm offer={offer} />);
+    wrapper.instance()['startDateChange'](undefined, start);
+    _.forEach(days, (day) => {
+      const end = day > 0 ? moment().add(day, 'days').toDate() : moment().subtract(day, 'days').toDate();
+      wrapper.instance()['endDateChange'](undefined, end);
+      expect(wrapper.find('.price').text()).to.contains((Math.abs(day) * price).toString());
+    });
+  });
+});
```
Comment (michalmikolajczyk) [inline src/ts/Detail/DetailForm.spec.tsx]: cool

File: src/ts/Detail/DetailForm.tsx [hunk 3]
```diff
@@ -111,6 +113,7 @@ export default class DetailForm extends React.Component<any, any> {
         <div className="field">
           <FontIcon className="fa fa-angle-down" />
           <SelectField
+            id="equipment"
```
Comment (michalmikolajczyk) [inline src/ts/Detail/DetailForm.tsx]: are you sure about this `id`? does it work well with React? Do we need ids?

File: src/ts/Detail/DetailHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+
+export default function DetailHeader(props) {
+  const { offer } = props;
+  return (
+    <div>
+      <div
+        className="image"
+        style={{ backgroundImage: `url(${offer.images.main})` }}
```
Comment (michalmikolajczyk) [inline src/ts/Detail/DetailHeader.tsx]: no spaces here

File: src/ts/Detail/DetailHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      />
+      <div className="title">
+        {offer.title}
+      </div>
+      <div className="owner">
+        <div
+          className="picture"
+          style={{ backgroundImage: `url(${offer.owner.picture})` }}
```
Comment (michalmikolajczyk) [inline src/ts/Detail/DetailHeader.tsx]: same

File: src/ts/Detail/DetailHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        />
+        <div className="owner-details">
+          <div className="fullname">
+            {offer.owner.fullname}
```
Comment (michalmikolajczyk) [inline src/ts/Detail/DetailHeader.tsx]: discuss inline text

File: src/ts/Detail/mockup.ts [hunk 7]
```diff
@@ -3,7 +3,7 @@ const ktm = {
   title: 'KTM SX 125, 2012',
   approx: '3 km away',
   type: 'Cross',
-  description: 'A brand new MX1 class two-stroke. Fantastic steering, light, and brings enough power to compete with 450cc four-stroke motocross bikes.',
+  description: 'Those wanting to race in the Junior World Championship need look no further than the 125 SX. The free-revving and lively two-stroke engine has gained even more performance for the current season.',
```
Comment (michalmikolajczyk) [inline src/ts/Detail/mockup.ts]: good

File: src/sass/Detail.scss (hunk 1 above)
Comment (jonzee) [inline src/sass/Detail.scss]: more specific class name

File: src/ts/Detail/DetailHeader.tsx (hunk 4 above)
Comment (jonzee) [inline src/ts/Detail/DetailHeader.tsx]: it's an object so it's ok to have space between brackets (airbnb style guide: https://github.com/airbnb/javascript/tree/master/react#quotes)

**Pull request merged**